tests/test_nmf_mfcc.py

```diff
@@ -0,0 +1,152 @@
+            value = np.load(file_path)
+            benchmark_dict[key] = value
+        return benchmark_dict
+
+    def test_mfcc_initialization(self):
+        # Load input file
+        input_file_name = os.path.join('..', 'input', 'piano_and_synth_arp_chord_mono.wav')
+        signal = nussl.AudioSignal(path_to_input_file=input_file_name)
+
+        # Set up NMMF MFCC using an int
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, mfcc_range=5, num_iterations=5)
+        assert nmf_mfcc.mfcc_start == 1
+        assert nmf_mfcc.mfcc_end == 5
+
+        nmf_mfcc.run()
+        sources = nmf_mfcc.make_audio_signals()
+
+        # Using a list
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, mfcc_range=[3, 15], num_iterations=5)
+        assert nmf_mfcc.mfcc_start == 3
+        assert nmf_mfcc.mfcc_end == 15
+
+        # Using a tuple
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, mfcc_range=(2, 14), num_iterations=5)
+        assert nmf_mfcc.mfcc_start == 2
+        assert nmf_mfcc.mfcc_end == 14
+
+    def test_random_seed_initialization(self):
+        # Load input file
+        input_file_name = os.path.join('..', 'input', 'piano_and_synth_arp_chord_mono.wav')
+        signal = nussl.AudioSignal(path_to_input_file=input_file_name)
+
+        # Differing random_seed and random_state, check if set separately
+        kmean_kwargs = {'random_state': 0}
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, num_iterations=5, random_seed=1,
+                                kmeans_kwargs=kmean_kwargs)
+        assert nmf_mfcc.clusterer.random_state == 0
+        assert nmf_mfcc.random_seed == 1
+
+        # No random_seed, but random_state initialized
+        kmean_kwargs = {'random_state': 0}
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, num_iterations=5, kmeans_kwargs=kmean_kwargs)
+        assert nmf_mfcc.clusterer.random_state == 0
+        assert nmf_mfcc.random_seed == None
+
+        # No random_state, but random_seed initialized
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, num_iterations=5, random_seed=0)
+        assert nmf_mfcc.clusterer.random_state == nmf_mfcc.random_seed == 0
+
+    def test_2_sin_waves(self):
+        sr = nussl.constants.DEFAULT_SAMPLE_RATE
+        dur = 3  # seconds
+        length = dur * sr
+        sine_wave_1 = np.sin(np.linspace(0, 440 * 2 * np.pi, length))
+        sine_wave_2 = np.sin(np.linspace(0, 440 * 5 * 2 * np.pi, length))
+        signal = sine_wave_1 + sine_wave_2
+        test_audio = nussl.AudioSignal()
+        test_audio.load_audio_from_array(signal)
+
+        # Set up NMMF MFCC
+        nmf_mfcc = nussl.NMF_MFCC(test_audio, num_sources=2, num_templates=2, num_iterations=15)
+
+        # and run
+        nmf_mfcc.run()
+        sources = nmf_mfcc.make_audio_signals()
+
+        for i, source in enumerate(sources):
```

Comment:
Tests shouldn't write to files. I'm assuming this test is just to make sure that nothing crashes, which is good (also a copy of the demo, also good). But exclude these last few lines where you write to output.